packages/ember-template-compiler/lib/plugins/utils.ts

```diff
@@ -12,25 +12,41 @@ export function isStringLiteral(node: AST.Expression): node is AST.StringLiteral
   return node.type === 'StringLiteral';
 }
 
+function getLocalName(node: string | AST.Expression | AST.VarHead) {
+  if (typeof node === 'string') return node;
+
+  if (node.type === 'VarHead') {
+    return node.original;
+  }
+
+  // surely this is wrong? (does it ever occur tho?)
+  // the type of params is `| Expression`, which isn't possible for block params
+  return node;
+}
+
 export function trackLocals() {
   let locals = new Map();
 
   let node = {
-    enter(node: AST.Program | AST.Block | AST.ElementNode) {
-      for (let param of node.blockParams) {
+    enter(node: AST.Template | AST.Block | AST.ElementNode | AST.BlockStatement) {
+      let params = 'params' in node ? node.params : node.blockParams;
```

Comment:
All the nodes you checked here should have `params`. Why this check? Unless you intend for this `enter` hook to be more generic, in which case you need to type `node: AST.Node` and do a different check here, because there are other things with a `params` property that is unrelated.

Reply:
`params does not exist on type Template` - we can play with removing it I guess, but I feel like it was needed for something and I don't remember -- I'll see if I rediscover that tomorrow

Reply:
oh yeah, that’s right, cause we don’t have locs for the js scope locals 👍